3498/3/app/controllers/users_controller.rb

```diff
@@ -0,0 +1,39 @@
+class UsersController < Sinatra::Base
+  include BCrypt
+  include UsersHelper
+
+  get '/signup' do
+    erb :signup
+  end
+
+  post '/registrations' do
+    @user = User.new(name: params[:username], email: params[:email], password: params[:password])
+    params[:password] ? @user.save : session[:error] = 'Enter password'
+    session[:user_id] = @user.id.to_s
```

Comment:
it will fail if user is not created. Try to process positive and negative flows in separate blocks